server/datastore/mysql/apple_mdm.go

```diff
@@ -1129,26 +1125,76 @@ INSERT INTO host_mdm_apple_profiles (
 		AND ds.profile_id IS NULL AND ds.uuid IS NULL
 		-- except "remove" operations in any state
 		AND ( hmap.operation_type IS NULL OR hmap.operation_type != ? )
-	) AS dt
+		-- profiles that are being installed
+	`
+
+		inArgs := []any{uuids, uuids, fleet.MDMAppleOperationTypeRemove}
+		if len(installIdentifiers) > 0 {
+			profilesToRemoveStmt += `AND hmap.profile_identifier NOT IN (?)`
+			inArgs = append(inArgs, installIdentifiers)
+
+		}
 
+		stmt, args, err = sqlx.In(profilesToRemoveStmt, inArgs...)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "building profiles to remove statement")
+		}
+		var profilesToRemove []*fleet.MDMAppleProfilePayload
+		err = sqlx.SelectContext(ctx, tx, &profilesToRemove, stmt, args...)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "bulk set pending profile status execute")
+		}
+
+		if len(profilesToInstall) == 0 && len(profilesToRemove) == 0 {
+			return nil
+		}
+
+		// before doing the inserts, remove profiles with identifiers that will be sent again from the database
+		stmt, args, err = sqlx.In("DELETE FROM host_mdm_apple_profiles WHERE host_uuid IN (?) AND profile_identifier IN (?)", uuids, installIdentifiers)
```

Comment:
I think that Martin's [comment](https://github.com/fleetdm/fleet/pull/10998#discussion_r1158842578) on a similar part of the other PR may apply here as well?

Reply:
that's a great point, thanks!